From d7d6bfc070a636c24b1ba33287d9ada8d7fdbd11 Mon Sep 17 00:00:00 2001
From: LuK1337 <priv.luk@gmail.com>
Date: Thu, 23 Aug 2018 22:34:12 +0200
Subject: [PATCH 6/8] lineage-sdk: Don't rely on EXTRA_WIFI_INFO when getting
 current SSID

* Nothing sets it anymore therefore it's always null.

Change-Id: I852e03acfd6fce9f51437cf5ad22b687ef1f083f
---
 .../lineageos/platform/internal/ProfileTriggerHelper.java    | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/lineage/lib/main/java/org/lineageos/platform/internal/ProfileTriggerHelper.java b/lineage/lib/main/java/org/lineageos/platform/internal/ProfileTriggerHelper.java
index 0c4f861..de466dc 100644
--- a/lineage/lib/main/java/org/lineageos/platform/internal/ProfileTriggerHelper.java
+++ b/lineage/lib/main/java/org/lineageos/platform/internal/ProfileTriggerHelper.java
@@ -115,10 +115,9 @@ public class ProfileTriggerHelper extends BroadcastReceiver {
                         Profile.TriggerState.ON_DISCONNECT);
                 mLastConnectedSSID = WifiSsid.NONE;
             } else if (NetworkInfo.DetailedState.CONNECTED.equals(state)) {
-                WifiInfo wifiInfo = intent.getParcelableExtra(WifiManager.EXTRA_WIFI_INFO);
-                WifiSsid ssid = wifiInfo.getWifiSsid();
+                String ssid = getActiveSSID();
                 if (ssid != null) {
-                    mLastConnectedSSID = ssid.toString();
+                    mLastConnectedSSID = ssid;
                     checkTriggers(Profile.TriggerType.WIFI, mLastConnectedSSID,
                             Profile.TriggerState.ON_CONNECT);
                 }
-- 
2.17.1

